homeassistant/components/mqtt/client.py

```diff
@@ -657,36 +656,31 @@ async def _async_perform_subscriptions(self) -> None:
         # Since we do not know if a published value is retained we need to
         # (re)subscribe, to ensure retained messages are replayed
 
-        def _process_client_subscriptions() -> list[tuple[int, int]]:
+        if not self._pending_subscriptions:
+            return
+
+        subscriptions: dict[str, int] = self._pending_subscriptions
+        self._pending_subscriptions = {}
+
+        def _process_client_subscriptions() -> tuple[int, int]:
             """Initiate all subscriptions on the MQTT client and return the results."""
-            subscribe_result_list = []
+            subscribe_result_list: list[tuple[int, int]] = []
```

Comment:
Not sure why this exists anymore.  Aside from appending to it, you're never using it (as you now only ever have 1 result and 1 mid).

Reply:
It is something of they old code and might be cleaned up.